libraries/botframework-schema/src/teams/index.ts

```diff
@@ -1225,6 +1229,167 @@ export interface FileConsentCardResponse {
     uploadInfo?: FileUploadInfo;
 }
 
+/**
+ * @interface
+ * Current tab request context, i.e., the current theme.
+ */
+export interface TabContext {
+    /**
+     * @member {theme} [string] The current user's theme.
+     */
+    theme?: string;
+}
+
+/**
+ * @interface
+ * Current TabRequest entity context, or 'tabEntityId'.
+ *
+ */
+export interface TabEntityContext {
+    /**
+     * @member {string} [tabEntityId] The entity id of the tab.
+     */
+    tabEntityId?: string;
+}
+
+/**
+ * @interface
+ * Invoke ('tab/fetch') request value payload.
+ *
+ */
+export interface TabRequest {
+    /**
+     * @member {TabEntityContext} [tabContext] The current tab entity request context.
+     */
+    tabContext?: TabEntityContext;
+    /**
+     * @member {TabContext} [context] The current user context, i.e., the current theme.
+     */
+    context?: TabContext;
+    /**
+     * @member {string} [state] The magic code for OAuth flow.
+     */
+    state?: string;
+}
+
+/**
+ * @interface
+ * Envelope for Card Tab Response Payload.
+ *
+ */
+export interface TabResponse {
+    /**
+     * @member {TabResponsePayload} [tab] The response to the tab/fetch message.
+     */
+    tab?: TabResponsePayload;
+}
+
+/**
+ * @interface
+ * Envelope for cards for a Tab request.
+ *
+ */
+export interface TabResponseCard {
+    /**
+     * @member {Record<string, any>} [card] The adaptive card for this card tab response.
+     */
+    card?: Record<string, unknown>;
+}
+
+/**
+ * @interface
+ * Envelope for cards for a TabResponse.
+ *
+ */
+export interface TabResponseCards {
+    /**
+     * @member {Record<string, any>[]} [cards] Adaptive cards for this card tab response.
+     */
+    cards?: Record<string, unknown>[];
+}
+
+/**
+ * @interface
+ * Payload for Tab Response.
+ *
+ */
+export interface TabResponsePayload {
+    /**
+     * @member {'continue' | 'auth' | 'silentAuth'} [type] Choice of action options when responding to the tab/fetch message.
+     */
+    type?: 'continue' | 'auth' | 'silentAuth';
```

Comment:
Not sure if this is intentional or not, are we departing from the current pattern of using a TypeScript `type` for Teams-specific interfaces with `type` members?

https://github.com/microsoft/botbuilder-js/blob/07ba1ea22cdb949e0230102bcdc6441871d7fed5/libraries/botframework-schema/src/teams/index.ts#L1659

https://github.com/microsoft/botbuilder-js/blob/07ba1ea22cdb949e0230102bcdc6441871d7fed5/libraries/botframework-schema/src/teams/index.ts#L1024-L1028

Reply:
IMO extracting a union is useful if you intend to write methods the accept parameters or return results typed as that union. Otherwise it feels unnecessary.

Reply:
@joshgummersall  So, leave this as-is?

Reply:
Yeah, I think it's fine for now. We can always extract it later if it becomes necessary.